Signal/src/ViewControllers/BlockListViewController.m

```diff
@@ -137,18 +137,7 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 - (NSString *)displayNameForIndexPath:(NSIndexPath *)indexPath
 {
     NSString *phoneNumber = _blockedPhoneNumbers[(NSUInteger)indexPath.item];
-    PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:phoneNumber];
-
-    // Try to parse and present the phone number in E164.
-    // It should already be in E164, so this should always work.
-    // If an invalid or unparsable phone number is already in the block list,
-    // present it as-is.
-    NSString *displayName = (parsedPhoneNumber ? parsedPhoneNumber.toE164 : phoneNumber);
-    Contact *contact = self.contactMap[displayName];
-    if (contact && [contact fullName].length > 0) {
-        displayName = [contact fullName];
-    }
-
+    NSString *displayName = [_contactsManager displayNameForPhoneIdentifier:phoneNumber];
```

Comment:
Ah, the _old_ 146.  

> Now in that case, I think we'll get nil.

Sorry, I still don't see the issue.  We're now using `displayNameForPhoneIdentifier`.  It doesn't try to parse the phone number - it just looks up a contact by phone number.  If no contact is found, it returns the phone number, not nil.  Isn't that the desired behavior?

```
- (NSString * _Nonnull)displayNameForPhoneIdentifier:(NSString * _Nullable)identifier {
    if (!identifier) {
        return self.unknownContactName;
    }
    Contact *contact = [self contactForPhoneIdentifier:identifier];
    
    NSString *displayName = (contact.fullName.length > 0) ? contact.fullName : identifier;
    
    return displayName;
}
```